### PR TITLE
Billy walker code review

### PR DESCRIPTION
**Pull request by StuartCowley**

### CODE REVIEW

File: src/kata1.fizzBuzz.js [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+    return "FizzBuzz";
+  }
+
+  if (isDivisableBy5) {
+    return "Buzz";
+  }
+  if (isDivisableBy3) {
+    return "Fizz";
+  }
+  return number;
```
Comment (StuartCowley) [inline src/kata1.fizzBuzz.js:14]: Id space this out a bit so it feels less cluttered
```
  const isDivisableBy5 = number % 5 === 0;

  if (isDivisableBy5 && isDivisableBy3) {
    return "FizzBuzz";
  }

  if (isDivisableBy5) {
    return "Buzz";
  }

  if (isDivisableBy3) {
    return "Fizz";
  }

  return number;
```

File: src/kata2.booleanToWord.js [hunk 2]
```diff
@@ -1,3 +1,10 @@
-const booleanToWord = (boolean) => {};
+const booleanToWord = (boolean) => {
+  if (boolean === true) {
+    return "yes";
+  }
+  if (boolean === false) {
+    return "no";
+  }
```
Comment (StuartCowley) [inline src/kata2.booleanToWord.js:7]: This works, but it could be made more concise using a ternary operator:
```
  return (boolean === true) ? "yes" : "no";
```
To go a little bit further, this could all be put on one line like this:
```
const booleanToWord = (boolean) => (boolean === true ? "yes" : "no");
```
This is the exact same as above, as when putting a function expression on one line like this you no longer need to use the `return` keyword as it is automatically implied

File: src/kata3.numberToReversedDigits.js [hunk 3]
```diff
@@ -1,3 +1,7 @@
-const numberToReversedDigits = (number) => {};
+const numberToReversedDigits = (number) => {
+  const numArray = number.toString().split('').map(Number);
+  const reversedArray = numArray.reverse();
+  return reversedArray;
```
Comment (StuartCowley) [inline src/kata3.numberToReversedDigits.js:4]: Theres no need to create the separate variables here, you can just chain the methods together.
Also, I wouldnt recommend using `map(Number)` like this in larger apps. While it works here, its better to better to be explicit with things like this (see [this stack overflow post](https://stackoverflow.com/questions/48343478/what-does-mapnumber-do-here))

Leaving the logic as it is you could update this to
```
const numberToReversedDigits = (number) =>
  number.toString().split("").map(Number).reverse();
```

File: src/kata4.humanCatDogYears.js [hunk 4]
```diff
@@ -1,3 +1,28 @@
-const humanCatDogYears = (number) => {};
+const humanCatDogYears = (number) => {
+  function calculateCatYears(humanYears) {
```
Comment (StuartCowley) [inline src/kata4.humanCatDogYears.js:2]: Id update this to use ES6 syntax:
```
  const calculateCatYears = (humanYears) => {
```

File: src/kata4.humanCatDogYears.js [hunk 5]
```diff
@@ -1,3 +1,28 @@
...
+    if (humanYears === 1) {
+      return 15;
+    }
+    if (humanYears === 2) {
+      return 24;
+    }
+    return 24 + (humanYears - 2) * 4;
+  }
+  function calculateDogYears(humanYears) {
```
Comment (StuartCowley) [inline src/kata4.humanCatDogYears.js:11]: As above, and Id add a clear line between the 2 functions for readability:
```
  }

  const calculateDogYears = (humanYears) => {
```

File: src/kata4.humanCatDogYears.js [hunk 6]
```diff
@@ -1,3 +1,28 @@
...
+    if (humanYears === 2) {
+      return 24;
+    }
+    return 24 + (humanYears - 2) * 5;
+  }
+
+  const catYears = calculateCatYears(number);
+  const dogYears = calculateDogYears(number);
+  const resultArray = [number, catYears, dogYears];
+  return resultArray;
```
Comment (StuartCowley) [inline src/kata4.humanCatDogYears.js:24]: No need to declare these extra variables here, just return the array with the function calls built in:
```
  return [number, calculateCatYears(number), calculateDogYears(number)];
```

File: src/kata5.reachDestination.js [hunk 7]
```diff
@@ -1,3 +1,7 @@
-const reachDestination = (distance, speed) => {};
+// eslint-disable-next-line arrow-body-style
```
Comment (StuartCowley) [inline src/kata5.reachDestination.js:1]: These exception doesnt need to be here, it should be taken out

File: src/kata6.joinNames.js [hunk 8]
```diff
@@ -1,3 +1,18 @@
-const joinNames = (namesObj) => {};
+// eslint-disable-next-line arrow-body-style
```
Comment (StuartCowley) [inline src/kata6.joinNames.js:1]: No need for the exception here, should take this out

File: src/kata6.joinNames.js [hunk 9]
```diff
@@ -1,3 +1,18 @@
...
+  }
+  if (namesObj.length === 2) {
+    namesString = namesOnly.join(" & ")
+  }
+  if (namesObj.length > 2) {
+    namesString = namesOnly.slice(0, -1).join(", ") + " & " + namesOnly.slice(-1);
+  }
+
+  return namesString;
+};
```
Comment (StuartCowley) [inline src/kata6.joinNames.js:17]: Nice approach! We can refactor this a bit by returning early whenever one of the if conditions is met, eg
```
  if (namesOnly.length === 1) {
    return namesOnly[0];
  }
```
This would mean we could lose thee variable on line 4
We could also remove the last if statement and use that as the default return if the other two havent been met by the time the work hits that. Putting that together, wee end up with this:
```
const joinNames = (namesObj) => {
  const namesOnly = namesObj.map((obj) => obj.name);

  if (namesOnly.length === 1) {
    return namesOnly[0];
  }

  if (namesObj.length === 2) {
    return namesOnly.join(" & ");
  }

  return `${namesOnly.slice(0, -1).join(", ")} & ${namesOnly.slice(-1)}`;
};

```

File: src/kata7.getEmployerRole.js [hunk 10]
```diff
@@ -1,3 +1,7 @@
-const getEmployerRole = (employeeName, employees) => {};
+// eslint-disable-next-line arrow-body-style
+const getEmployerRole = (employeeName, employees) => {
+  const findRole = employees.find(({ name }) => name === employeeName);
+  return findRole.role;
```
Comment (StuartCowley) [inline src/kata7.getEmployerRole.js:4]: This could be returned directly in one line:
```
  return employees.find(({ name }) => name === employeeName).role;
```

File: src/kata7.getEmployerRole.js [hunk 11]
```diff
@@ -1,3 +1,7 @@
-const getEmployerRole = (employeeName, employees) => {};
+// eslint-disable-next-line arrow-body-style
```
Comment (StuartCowley) [inline src/kata7.getEmployerRole.js:1]: No need for the exception here!

File: test/kata3.numberToReversedDigits.test.js [hunk 12]
```diff
@@ -1,5 +1,15 @@
 const { numberToReversedDigits } = require("../src");
 
 describe("numberToReversedDigits", () => {
-  test("returns a reversed array of the number's digits", () => {});
-});
\ No newline at end of file
+  test("returns 12345 arrayed and reversed", () => {
```
Comment (StuartCowley) [inline test/kata3.numberToReversedDigits.test.js:4]: Could any tests for shorter arrays be added, say `[2, 3]`? Just to cover more use cases

File: src/kata4.humanCatDogYears.js [hunk 13]
```diff
@@ -1,3 +1,28 @@
-const humanCatDogYears = (number) => {};
+const humanCatDogYears = (number) => {
```
Comment (StuartCowley) [inline src/kata4.humanCatDogYears.js:1]: Would happen if 0 was entered? We could add a test for that...
That would be an edge case we should take into account in the code. We can do that by adding an extra clause at the start, eg
```
  if (number === 0) {
    return [0, 0, 0];
  }
```

File: test/kata4.humanCatDogYears.test.js [hunk 14]
```diff
@@ -1,3 +1,15 @@
 const { humanCatDogYears } = require("../src");
 
-// Look Ma, no handlebars!!!
+describe("humanCatDogYears", () => {
+  it("returns array of 10, 56, 64", () => {
```
Comment (StuartCowley) [inline test/kata4.humanCatDogYears.test.js:4]: A test for passing in 0 would be good here

File: test/kata6.joinNames.test.js [hunk 15]
```diff
@@ -1,5 +1,41 @@
 const { joinNames } = require("../src");
 
 describe("joinNames", () => {
-  test("returns string of names, seperated by commas and an ampersand", () => {});
+  // eslint-disable-next-line prettier/prettier
```
Comment (StuartCowley) [inline test/kata6.joinNames.test.js:4]: Why disable prettier here? I would remove all these and let the formatter do its job

File: test/kata7.getEmployerRole.test.js [hunk 16]
```diff
@@ -1,5 +1,33 @@
...
+    {
+      name: "Jenny",
+
+      role: "Sales Associate",
+    },
+    {
+      name: "Javid",
+
+      role: "Human Recommended Reading Assistant",
+    },
```
Comment (StuartCowley) [inline test/kata7.getEmployerRole.test.js:19]: No need for the extra spaces here:
```
    {
      name: "Satti",
      role: "Developer",
    },
    {
      name: "Jenny",
      role: "Sales Associate",
    },
    {
      name: "Javid",
      role: "Human Recommended Reading Assistant",
    },
```

File: test/kata7.getEmployerRole.test.js [hunk 17]
```diff
@@ -1,5 +1,33 @@
...
+  ];
+  it("returns Javid's role", () => {
+    expect(getEmployerRole("Javid", employees)).toBe(
+      "Human Recommended Reading Assistant"
+    );
+  });
+  it("returns Jenny's role", () => {
+    expect(getEmployerRole("Jenny", employees)).toBe("Sales Associate");
```
Comment (StuartCowley) [inline test/kata7.getEmployerRole.test.js:27]: Add a clear line between each of your tests for better readability:
```
  ];

  it("returns Javid's role", () => {
    expect(getEmployerRole("Javid", employees)).toBe(
      "Human Recommended Reading Assistant"
    );
  });

  it("returns Jenny's role", () => {
    expect(getEmployerRole("Jenny", employees)).toBe("Sales Associate");
```

**Review by StuartCowley — COMMENTED** [review]
Good work! Everything passes and things are mostly well formatted and organised, and your tests cover good ground of the code too
Have left some comments about the eslint/prettier exceptions which arent needed, some more readability improvements, plus some refactoring to tidy things up a bit and make them, more concise

Good stuff!
